test/spec/controllers/registration.spec.js

```diff
@@ -0,0 +1,110 @@
+import angular from 'angular';
+import 'angular-mocks';
+
+describe('Controller: registration', () => {
+  let scope, testData;
+
+  beforeEach(angular.mock.module('confRegistrationWebApp'));
+
+  beforeEach(
+    angular.mock.inject(
+      (
+        $rootScope,
+        $controller,
+        $routeParams,
+        _testData_,
+        _validateRegistrant_,
+      ) => {
+        testData = _testData_;
+        scope = $rootScope.$new();
+        scope.conference = testData.conference;
+        scope.currentRegistration = testData.registration;
+        angular.extend($routeParams, {
+          reg: testData.registration.registrants[0].id,
+          pageId: testData.conference.registrationPages[0].id,
+        });
+
+        $controller('RegistrationCtrl', {
+          $rootScope,
+          $scope: scope,
+          conference: testData.conference,
+          currentRegistration: testData.registration,
+          $routeParams,
+          validateRegistrant: _validateRegistrant_,
+        });
+      },
+    ),
+  );
+
+  it('should have validPages based on current registrant', () => {
+    let validPages = scope.validPages;
+
+    expect(validPages.length).toEqual(2);
+  });
+
+  it('should getFirstValidPage', () => {
+    let current = scope.currentRegistrant;
```

Comment:
A bit nit-picky from my, but `currentRegistrant` would be better than `current`.